spec/puppetserver/ca/local_certificate_authority_spec.rb

```diff
@@ -1,23 +1,42 @@
 require 'puppetserver/ca/local_certificate_authority'
+require 'puppetserver/ca/config/puppet'
 
+require 'utils/ssl'
 require 'puppetserver/ca/utils/signing_digest'
 
 RSpec.describe Puppetserver::Ca::LocalCertificateAuthority do
+  include Utils::SSL
 
   let(:settings) {
-    { :ca_ttl => 157680000,
-      :ca_name => 'bulla2',
-      :subject_alt_names => '',
-      :root_ca_name => 'bulla',
-      :certname => 'ulla',
-      :keylength => 512,
-      :hostprivkey => '$privatekeydir/$certname.pem',
-      :hostpubkey => '$publickeydir/$certname.pem',
-      :csr_attributes => '$confdir/csr_attributes.yaml',
-      :serial => '$cadir/serial' } }
+    with_ca_in(Dir.mktmpdir) do |config|
+      return Puppetserver::Ca::Config::Puppet.new(config).load({})
+    end
```

Comment:
Oh nice, this seems way better. I wonder if there are other places where we could do something like this, for example in https://github.com/puppetlabs/puppetserver-ca-cli/blob/master/spec/puppetserver/ca/utils/http_client_spec.rb...

Reply:
I've updated the PR to have a commit that refactors that http_client_spec.